pkg/repo/mysql/services.go

```diff
@@ -0,0 +1,92 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.expect.digital/translate/pkg/model"
+	"go.expect.digital/translate/pkg/repo"
+)
+
+func (r *Repo) SaveService(ctx context.Context, service *model.Service) error {
+	query := `INSERT INTO Service (ID, Name) VALUES (?, ?)
+	ON DUPLICATE KEY UPDATE Name = VALUES(Name)`
+
+	_, err := r.db.ExecContext(ctx, query, service.ID, service.Name)
+	if err != nil {
+		return fmt.Errorf("db: insert service: %w", err)
+	}
+
+	return nil
+}
+
+func (r *Repo) LoadService(ctx context.Context, serviceID uuid.UUID) (*model.Service, error) {
+	query := `SELECT ID, Name FROM Service WHERE ID = ?`
+	row := r.db.QueryRowContext(ctx, query, serviceID)
+
+	var service model.Service
+
+	err := row.Scan(&service.ID, &service.Name)
+
+	switch {
+	case errors.Is(err, sql.ErrNoRows):
+		return nil, repo.ErrNotFound
+	case err != nil:
+		return nil, fmt.Errorf("db: select service: %w", err)
+	}
+
+	return &service, nil
```

Comment:
This is how I do it, you don't have to follow it. `default` case is at the top to ease the reading of the happy path, bool switch provides a clear view of different outcomes.

```suggestion
	err := row.Scan(&service.ID, &service.Name)
	switch {
	default:
		return &service, nil
	case errors.Is(err, sql.ErrNoRows):
		return nil, repo.ErrNotFound
	case err != nil:
		return nil, fmt.Errorf("db: select service: %w", err)
	}	
```

Reply:
okay, I am just used to `default` always being the last option in switch cases, not sure which option  I like best.

Reply:
I like it. It brings to attention what is the default outcome, followed by alternatives and unhappy paths. All outcomes are laid out in parallel - only one path is possible, with less code.